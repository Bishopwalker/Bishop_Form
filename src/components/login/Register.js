import React, {useState} from 'react'
import {Link} from 'react-router-dom';
import {useDispatch} from 'react-redux';
import {registerUser} from "../../ducks/reducer";
const Register = () => {
    let dispatch = useDispatch();
    const [formData, setFormData] = useState({
        firstname: '',
        email: '',
        password: '',
        password2: ''
    });

    const {firstname, email, password, password2} = formData;

    const onChange = e => setFormData({...formData, [e.target.name]: e.target.value});

    const onSubmit = async e => {
        e.preventDefault();
        if (password !== password2) {
            alert('Passwords do not match');
        } else {
            console.log('Register');
            dispatch(registerUser({firstname, email, password}));
        }
    };

    return (
        <div className="register">
            <div className="container">
                <div className="row">
                    <div className="col-md-8 m-auto">
                        <h1 className="display-4 text-center">Sign Up</h1>
                        <p className="lead text-center">Create your DevConnector account</p>
                        <form onSubmit={e => onSubmit(e)}>
                            <div className="form-group">
                                <input type="text" className="form-control form-control-lg" placeholder="First Name" name="firstname" value={firstname} onChange={e => onChange(e)}/>
                            </div>
                            <div className="form-group">
                                <input type="email" className="form-control form-control-lg" placeholder="Email Address" name="email" value={email} onChange={e => onChange(e)}/>
                            </div>
                            <div className="form-group">
                                <input type="password" className="form-control form-control-lg" placeholder="Password" name="password" value={password} onChange={e => onChange(e)}/>
                            </div>
                            <div className="form-group">
                                <input type="password" className="form-control form-control-lg" placeholder="Confirm Password" name="password2" value={password2} onChange={e => onChange(e)}/>
                                    </div> <button type="submit" className="btn btn-info btn-block mt-4">Submit</button>
                                </form>
                        <Link to="/login" className="btn btn-outline-info btn-block mt-4">Login</Link>
                    </div>
                                        </div>
                                            </div>
                                                </div>
    )
    }
        export default Register;