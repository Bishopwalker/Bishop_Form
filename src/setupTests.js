// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import '@testing-library/jest-dom';
import {render as rtlRender} from '@testing-library/react';
import {MemoryRouter} from 'react-router-dom'
import {configureStore} from '@reduxjs/toolkit';
import {Provider} from 'react-redux';
import reducer from './ducks/reducer'
import {BrowserRouter} from "react-router-dom";
import {createMemoryHistory} from "history";
//create a customRender that wraps the UI in a memory Router
// const customRender=(ui,options)=>{
//     return render(ui, {wrapper: MemoryRouter, ...options})
// }
// //re-export everything
// export * from '@testing-library/react'
//
// //override render method
//
// export {customRender as render};

function render(
    ui,
    {   route = '/',
        history = createMemoryHistory({initialEntries: [route]}),
        preloadedState,
        store = configureStore({ reducer: { ...reducer, ...preloadedState } }),
        ...renderOptions
    } ={}
) {
    function Wrapper({ children }) {
        return <Provider store={store}>{children}</Provider>;
    }
    return rtlRender(ui, { wrapper: Wrapper, ...renderOptions });
}
// re-export everything
export * from '@testing-library/react';

// override render method
export {render};
