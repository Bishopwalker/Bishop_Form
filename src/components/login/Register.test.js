import React from 'react';
import {render,cleanup,screen} from '../../setupTests'
import * as reactRedux  from 'react-redux';
import {BrowserRouter} from "react-router-dom";
import Register from './Register'

describe('Register person form',()=>{
    const useDispatchMock = jest.spyOn(reactRedux, 'useDispatch');

    beforeEach(() => {
        useDispatchMock.mockClear();
    });
test('Should take a snapshot',()=>{
    const {asFragment} = render(<BrowserRouter><Register/></BrowserRouter>)
    expect(asFragment()).toMatchSnapshot()
})
    test('Should Render the words signup',async()=>{
        await render(<Register/>,{wrapper: BrowserRouter})
        expect( screen.getByText(/Sign Up/i)).toBeInTheDocument()
    })
    test('Should have the words Create your Hassan Account',()=>{
        render(<Register/>,{wrapper: BrowserRouter})
        expect(screen.getByText(/Create your Hassan Account/i)).toBeInTheDocument()
    })
    test('Should have the words Email Address ',()=>{
        render(<Register/>,{wrapper: BrowserRouter})

        expect(screen.getByPlaceholderText(/Email Address/i)).toBeInTheDocument()
    })
    test('Should have a link with the word Login written',()=>{
        render(<Register/>,{wrapper: BrowserRouter})
        expect(screen.getByText(/Login/i)).toBeInTheDocument()
        expect(screen.getByRole('link',{name:/Login/i})).toBeInTheDocument()
    })
    test('Should have the words Password twice',async()=>{
        render(<Register/>,{wrapper: BrowserRouter})
        expect( screen.getAllByPlaceholderText(/Password/i)).toHaveLength(2)
    })
    test('Should have a placeholder for the word Confirm Password',async()=>{
        render(<Register/>,{wrapper: BrowserRouter})
        expect(await screen.findByPlaceholderText(/Confirm Password/i)).toBeInTheDocument()
    })
    test('Should have a button with the word Submit',async()=>{
        render(<Register/>,{wrapper: BrowserRouter})
        expect(await screen.findByRole('button',{name:/Submit/i})).toBeInTheDocument()
    })
})