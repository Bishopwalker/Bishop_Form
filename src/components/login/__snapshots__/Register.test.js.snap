// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Register person form Should take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="register"
  >
    <div
      class="container"
    >
      <div
        class="row"
      >
        <div
          class="col-md-8 m-auto"
        >
          <h1
            class="display-4 text-center"
          >
            Sign Up
          </h1>
          <p
            class="lead text-center"
          >
            Create your Hassan account
          </p>
          <form>
            <div
              class="form-group"
            >
              <input
                class="form-control form-control-lg"
                name="firstname"
                placeholder="First Name"
                type="text"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="form-control form-control-lg"
                name="email"
                placeholder="Email Address"
                type="email"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="form-control form-control-lg"
                name="password"
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
            <div
              class="form-group"
            >
              <input
                class="form-control form-control-lg"
                name="password2"
                placeholder="Confirm Password"
                type="password"
                value=""
              />
            </div>
             
            <button
              class="btn btn-info btn-block mt-4"
              type="submit"
            >
              Submit
            </button>
          </form>
          <a
            class="btn btn-outline-info btn-block mt-4"
            href="/login"
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
